lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/EvaluatorManager.java

```diff
@@ -487,6 +487,9 @@ public void onResourceLaunch(final ResourceLaunchEvent resourceLaunchEvent) {
       if (this.stateManager.isAllocated()) {
         this.stateManager.setSubmitted();
         this.resourceLaunchHandler.onNext(resourceLaunchEvent);
+      } else if (this.stateManager.isFailedOrKilled()) {
+        LOG.log(Level.WARNING, "Evaluator manager expected" + EvaluatorState.ALLOCATED +
```

Comment:
I would keep it as `WARNING`. 1. This occurs infrequently so is not noisy, and 2. If for some reason `FailedEvaluator` event is lost or not processed, this code path will secretly not launch task, and I feel it's better to leave some breadcrumbs for the investigation, rather than hide the decision completely.